### PR TITLE
refactor: make the concept of storage clients abstract

### PR DESCRIPTION
**Pull request by B4nan**
WIP

### CODE REVIEW

File: packages/apify/src/actor.ts [hunk 1]
```diff
@@ -1352,6 +1351,40 @@ export interface MetamorphOptions {
...
+     * ```
+     * {
+     *   "contentType": "application/json; charset=utf-8",
+     *   "body": {
+     *     "message": "Hello world!"
+     *   }
+     * }
+     * ```
+     */
+    output?: Dictionary | null;
```
Comment (mnmkng) [inline packages/apify/src/actor.ts:1387]: I would keep the `call` API exactly the same as Client now. The final version of how this whole output management will look like is still undecided. So there should be no need for this type.

Comment (B4nan) [reply packages/apify/src/actor.ts:1387]: This PR is not about that, the diff is just about moving types from a different package to the actor sdk.

But good catch, this is indeed forgotten type that is wrong (there is no `fetchOutput` anymore in the code), will remove it.

Comment (mnmkng) [reply packages/apify/src/actor.ts:1387]: Not sure I understand. You can't have docs for a method you don't have a separate interface for?

Comment (B4nan) [reply packages/apify/src/actor.ts:1387]: Sure we can, but the client type has no jsdoc at all for `ActorRun`, and it does not make much sense to have jsdoc talking about the `Actor` class from SDK in client.

Comment (mnmkng) [reply packages/apify/src/actor.ts:1387]: I think having the return value documented in the method will be enough.

**Pull request closed without merging**